File: src/main/java/chapter4/RangeExample.java
package chapter4;

import common.Log;
import io.reactivex.Observable;


public class RangeExample {

    public void forLoop() {
        Observable<Integer> source = Observable.range(1, 10)
                .filter(number -> number % 2 == 0);
        source.subscribe(Log::it);
    }

    public static void main(String[] args) {
        RangeExample demo = new RangeExample();
        demo.forLoop();
    }

}

File: src/main/kotlin/chapter4/create/RangeExample_.kt
package chapter4.create

import common.Log_
import io.reactivex.Observable


class RangeExample_ {

    fun forLoop() {
        val source = Observable.range(1, 10)
                .filter { it % 2 == 0 }
        source.subscribe { Log_.it(it) }
    }

}


fun main(args: Array<String>) {
    val demo = RangeExample_()
    demo.forLoop()
}
